fsmonitor.c

```diff
@@ -203,7 +207,7 @@ void refresh_fsmonitor(struct index_state *istate)
 	strbuf_release(&query_result);
 
 	/* Now that we've updated istate, save the last_update time */
-	istate->fsmonitor_last_update = last_update;
+	istate->fsmonitor_last_update = last_update - 1000000000;
```

Comment:
Please comment on the change here and why we are subtracting.

I changed the example in `templates/hooks--fsmonitor-watchman.sample` and `t/t7519/fsmonitor-watchman` instead, as they seem to be watchman-specific adjustments instead of something universal to all `fsmonitor` integrations.

Reply:
Also, `t/t7519/fsmonitor-watchman` has the following query:

```
["query", "$git_work_tree", {
			"since": $time,
			"fields": ["name"],
			"expression": ["not", ["allof", ["since", $time, "cclock"], ["not", "exists"]]]
		}]
```

but the template doesn't have the `expression` line. That seems to be the thing causing the add/delete nullifications, right?

Reply:
Yes that expression is what is filtering out the changes since X that no longer exist.

Reply:
I'll move some things and clean this up and put it on it's own PR so we can test without my other changes.

Reply:
(Sorry, I think I had the files backwards as to which has the expression and which does not.)